tests/test_app.py

```diff
@@ -0,0 +1,31 @@
+from model_service.app import app
+
+SAMPLE_URL = "https://sample_url.com"
+
+
+def test_response():
+    with app.test_client() as c:
+        response = c.post('/predict', json={"url": SAMPLE_URL})
+        # Check if the response code is correct
+        assert response.status_code == 200
+
+
+def test_keys():
+    with app.test_client() as c:
+        response = c.post('/predict', json={"url": SAMPLE_URL})
+        # Check if keys are correct
+        assert response.json.keys() == {"prediction_binary", "prediction"}
+
+
+def test_prediction_binary():
+    with app.test_client() as c:
+        response = c.post('/predict', json={"url": SAMPLE_URL})
+        # Check if values are correct
+        assert isinstance(response.json["prediction_binary"][0][0], int)
+
+
+def test_prediction():
+    with app.test_client() as c:
+        response = c.post('/predict', json={"url": SAMPLE_URL})
+        # Check if values are correct
+        assert isinstance(response.json["prediction"][0][0], float)
```

Comment:
Might as well put everything into one test with multiple asserts

Reply:
My rule of thumb is tests should have 3 parts:
1. Setup (mocks, renders, etc.)
2. Action (function calls, requests, user interaction)
3. Assertion

If the first 2 are the same you can combine the tests 😄

Reply:
I've setup the tests in different functions because in one function if an assert triggered, it wouldn't display which assert specifically failed the test. In the results it showed which function failed and not the assertion.

However if I am missing something with the implementation please let me know, maybe a setting or specific way to implement it so the assertions are in one test but it still reports which assertion fails?

I've merged it for now since this works, next pr I'll clean it up